common/src/main/java/com/microsoft/identity/common/CodeMarkerUtil.java

```diff
@@ -35,20 +35,19 @@
      * @return string to save
      */
     public static String getCsvContent(final List<CodeMarker> codeMarkers) {
-        if (codeMarkers == null) {
+        if (codeMarkers == null || codeMarkers.isEmpty()) {
             return "";
         }
 
-        final StringBuilder stringToWrite = new StringBuilder();
-        if (codeMarkers.size() > 0) {
-            stringToWrite.append(codeMarkers.get(0).getCsvHeader());
-        }
+        final StringBuilder content = new StringBuilder();
+        content.append(codeMarkers.get(0).getCsvHeader());
 
-        for (final CodeMarker codeMarker : codeMarkers) {
-            stringToWrite.append('\n');
-            stringToWrite.append(codeMarker.getCsvLine());
+        synchronized (codeMarkers) {
```

Comment:
This is interesting.  Is something else also synchronizing on this monitor?

Reply:
I thought of synchronizing on code markers so that any modifications at this point don't affect the list. I would also appreciate your advice on how best to handle this: [`markCode`](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/CodeMarkerManager.java#L63) adds the code markers, [`clearMarkers`](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/CodeMarkerManager.java#L117) removes all while [`getCsvContent`](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/c6bb5aaf9d377ba5a40bbdc224cceb81543881a3/common/src/main/java/com/microsoft/identity/common/CodeMarkerUtil.java#L46) has the loop that sometimes was failing with ConcurrentModificationException

Reply:
OK, this method is counting on receiving a SynchronizedList, so that the synchronization on the list itself blocks all other accesses - we should document that, since any other user will not know it if that's what we're after.  Alternately, since this is all called from within the CodeMarkerManager file, we could just synchronize those methods, rather than having them synchronize internally on the list reference.

Is it possible that 1/2 of a pair or markers could be added while this process goes on?

Is there any reason that this is in a separate class?